src/frontend/actors_generation_phase.nit

```diff
@@ -0,0 +1,261 @@
+			# Used if the call must return a value
+			var ret_val = ""
+			var ret_val2 = ""
+
+			# Params to send to `instance` in the `invoke` method
+			var params = ""
+
+			# Values for the generated proxy method
+			# Return signature
+			var ret = ""
+			# Return statement
+			var ret2 = ""
+
+			if signature != null then
+				var proxy_params= new Array[String]
+
+				# Deal with parameters
+				var mparameters = signature.mparameters
+				if mparameters.length > 0 then
+					var parameters = new Array[String]
+					proxy_sign += "("
+					for p in mparameters do
+						var n = p.name
+						var t = p.mtype.name
+						msg_attributes.add("var " + n + ": " + t)
+						proxy_params.add(n + ": " + t)
+						parameters.add(n)
+					end
+					proxy_sign += proxy_params.join(", ") + ")"
+					params = "(" + parameters.join(", ") + ")"
+				end
+
+				# Deal with the return if any
+				var ret_type = signature.return_mtype
+				if ret_type != null then
+					msg_attributes.add("var ret = new Future[{ret_type.name}]")
+					ret_val = "ret.set_value("
+					ret_val2 = ")"
+					ret = ": Future[{ret_type.name}]"
+					ret2 = "return msg.ret"
+				end
+			end
+
+			# Name of the Message class
+			var name = classname + "Message" + method_name
+
+			# The effective Message Class
+			messages.add(
+"""
+class {{{name}}}
+	super {{{msg_class_name}}}
+
+	{{{msg_attributes.join("\n")}}}
+
+	redef fun invoke(instance) do {{{ret_val}}}instance.{{{method_name}}}{{{params}}}{{{ret_val2}}}
+end
+""")
+
+
+# The actual proxy call
+			proxy_calls.add(
+"""
+redef fun {{{method_name}}}{{{proxy_sign}}}{{{ret}}} do
+	var msg = new {{{name}}}{{{params}}}
+	actor.mailbox.push(msg)
+	{{{ret2}}}
+end
+""")
+		end
+
+		# At this point, all msg classes should be good
+		# All of the functions of the proxy too
+
+		# Let's generate the proxy class then
+		proxys.add(
+"""
+redef class Proxy{{{classname}}}
+
+	redef type E: Actor{{{classname}}}
+	#var actor: Actor{{{classname}}} is noinit
+
+	init proxy(base_class: {{{classname}}}) do
+		actor = new Actor{{{classname}}}(base_class)
+		actor.start
+	end
+
+	{{{proxy_calls.join("\n\n")}}}
+end
+""")
+
+		redef_classes.add(
+"""
+redef class {{{classname}}}
+redef var async is lazy do return new Proxy{{{classname}}}.proxy(self)
+end
+""")
+	end
+
+	redef fun process_nmodule_after(nmodule) do
+		var first_mmodule = nmodule.mmodule
+		assert first_mmodule != null
+
+		# Be careful not to generate useless submodules !
+		if not first_mmodule.generate_actor_submodule then return
+
+		# Name of the support module
+		var module_name
+
+		# Path to the support module
+		module_name = "actors_{first_mmodule.name}"
+
+		# We assume a module using actors has a `filepath` not null
+		var mmodule_path = first_mmodule.filepath.as(not null).dirname
+
+		var module_path = "{mmodule_path}/{module_name}.nit"
+
+		var nit_module = new NitModule(module_name + " is no_warning(\"missing-doc\")")
```

Comment:
You should set the modules annotations with `nit_module.annotations.add "no_warning..."`.